lib/membrane_rtmp_plugin/rtmp_server/config.ex

```diff
@@ -0,0 +1,363 @@
+
+      {certfile, keyfile} when is_binary(certfile) and is_binary(keyfile) ->
+        validate_cert_and_key_files(opts, validate_files, certfile, keyfile)
+
+      {certfile, nil} when is_binary(certfile) ->
+        validate_single_cert_file(
+          validate_files,
+          "SSL certificate file provided but key file is missing"
+        )
+
+        opts
+
+      {nil, keyfile} when is_binary(keyfile) ->
+        validate_single_cert_file(
+          validate_files,
+          "SSL key file provided but certificate file is missing"
+        )
+
+        opts
+    end
+  end
+
+  @spec validate_cert_and_key_files(ssl_options(), boolean(), String.t(), String.t()) ::
+          ssl_options()
+  defp validate_cert_and_key_files(opts, validate_files, certfile, keyfile) do
+    if validate_files do
+      validate_file_exists(certfile, "SSL certificate")
+      validate_file_exists(keyfile, "SSL key")
+      validate_additional_cert_files(opts)
+    end
+
+    opts
+  end
+
+  @spec validate_additional_cert_files(ssl_options()) :: :ok
+  defp validate_additional_cert_files(opts) do
+    if opts[:cacertfile], do: validate_file_exists(opts[:cacertfile], "SSL CA certificate")
+    if opts[:certchain], do: validate_file_exists(opts[:certchain], "SSL certificate chain")
+    :ok
+  end
+
+  @spec validate_single_cert_file(boolean(), String.t()) :: :ok
+  defp validate_single_cert_file(validate_files, error_message) do
+    if validate_files do
+      raise ArgumentError, error_message
+    end
+
+    :ok
+  end
+
+  @spec validate_file_exists(String.t(), String.t()) :: :ok
+  defp validate_file_exists(file_path, file_type) do
+    unless File.exists?(file_path) do
+      raise ArgumentError, "#{file_type} file does not exist: #{file_path}"
+    end
+
+    :ok
+  end
+
+  @spec validate_ssl_configuration(ssl_options()) :: ssl_options()
+  defp validate_ssl_configuration(opts) do
+    # Validate TLS versions
+    if versions = opts[:versions] do
+      valid_versions = [:tlsv1, :"tlsv1.1", :"tlsv1.2", :"tlsv1.3"]
+      invalid_versions = versions -- valid_versions
+
+      unless Enum.empty?(invalid_versions) do
+        raise ArgumentError,
+              "Invalid TLS versions: #{inspect(invalid_versions)}. " <>
+                "Valid versions are: #{inspect(valid_versions)}"
+      end
+    end
+
+    # Validate verify option
+    if verify = opts[:verify] do
+      unless verify in [:verify_none, :verify_peer] do
+        raise ArgumentError,
+              "Invalid verify option: #{inspect(verify)}. " <>
+                "Must be :verify_none or :verify_peer"
+      end
+    end
+
+    # Validate log level
+    if log_level = opts[:log_level] do
+      valid_levels = [:none, :error, :warning, :notice, :info, :debug, :all]
+
+      unless log_level in valid_levels do
+        raise ArgumentError,
+              "Invalid SSL log level: #{inspect(log_level)}. " <>
+                "Valid levels are: #{inspect(valid_levels)}"
+      end
```

Comment:
do we really need to validate it here? I would leave it up to the `:ssl` Erlang module